src/mount-support.c

```diff
@@ -293,3 +294,50 @@ void setup_slave_mount_namespace()
 		die("can not make make / rslave");
 	}
 }
+
+void setup_bind_mounts(const char *appname)
+{
+	debug("%s: %s", __FUNCTION__, appname);
+
+	FILE *f = NULL;
+	const char *bind_profile_dir = "/var/lib/snapd/bind/profiles/";
+
+	char profile_path[PATH_MAX];
+	must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bind",
+		      bind_profile_dir, appname);
+
+	f = fopen(profile_path, "r");
```

Comment:
Can you please use `__attribute__((cleanup(sc_cleanup_file)))` and add `sc_close_file` to `cleanup-funcs.[ch]` with something like this inside:

```
void sc_cleanup_file(f **FILE) {
    if (*f != NULL) {
        fclose(*f);
    }
}
```

This should simplify the logic below.